### PR TITLE
feat(Core/Events): update holiday code and remove misleading log (event date is the one from game_event)

### LINKED ISSUE #3360: No suitable start date found for holiday XXX
No suitable start date found for holiday XXX

##### CURRENT BEHAVIOUR:

When starting up worldserver.exe there are 6 Errors (? check if they are real errors by definition) talking about "no suitable start dates found for holidays XXX. 

Those holidays are the 6 from the Call to Arms events.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

I guess there shouldnt be any error message.

##### STEPS TO REPRODUCE THE PROBLEM:

1.  Well just start-up worldserver.exe
2.  Scroll up until you see the errors.

##### EXTRA NOTES:

Works on every other platform I Have used (CMaNGOS, TC and Warmane and such)

##### AC HASH/COMMIT:

https://github.com/azerothcore/azerothcore-wotlk/commit/d8ad0c520060280fe48fb9d73e2284cad241794d

##### OPERATING SYSTEM:

Windows 10

##### MODULES:

No

##### OTHER CUSTOMIZATIONS:

No

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/92512246-no-suitable-start-date-found-for-holiday-xxx?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Helias**
Cherry pick from https://github.com/TrinityCore/TrinityCore/commit/5b4287e6832da32539a49e68f6a5ab0ee55854b3

## ISSUES ADDRESSED:
- Closes #3360

## TESTS PERFORMED:
- I didn't test

## HOW TO TEST THE CHANGES:
https://github.com/azerothcore/azerothcore-wotlk/issues/3360

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Helias** [comment]
@Kitzunu 
I ported the TC code to AC, but I am not sure why TC still send a uint32() to the client instead of a Int32().
Check CalendarHandler.cpp, look at `holiday->CalendarFilterType`.
I didn't touch it because probably the client does not support a negative value for this.

File: src/server/shared/DataStores/DBCStructure.h [hunk 1]
```diff
@@ -1075,7 +1075,7 @@ struct HolidaysEntry
     //uint32 holidayDescriptionId;                          // 50       m_holidayDescriptionID (HolidayDescriptions.dbc)
     char* TextureFilename;                                  // 51       m_textureFilename
     uint32 Priority;                                        // 52       m_priority
-    uint32 CalendarFilterType;                              // 53       m_calendarFilterType (-1 = Fishing Contest, 0 = Unk, 1 = Darkmoon Festival, 2 = Yearly holiday)
+    int32 CalendarFilterType;                              // 53       m_calendarFilterType (-1 = Fishing Contest, 0 = Unk, 1 = Darkmoon Festival, 2 = Yearly holiday)
```
Comment (Kitzunu) [inline src/server/shared/DataStores/DBCStructure.h:1078]:
```suggestion
    int32 CalendarFilterType;                               // 53       m_calendarFilterType (-1 = Fishing Contest, 0 = Unk, 1 = Darkmoon Festival, 2 = Yearly holiday)
```

**Comment by Kitzunu** [comment]
Actually I dont know why... seems weird that they would change it in only one location, or not keep it as it was :shrug:
I will see if I can figure something out

**Comment by Kitzunu** [comment]
According to the original commit message:
- kill localtime usage as it's deprecated

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 2]
```diff
@@ -29,6 +29,18 @@
 #include "LuaEngine.h"
 #endif
 
+#if !HAVE_TIME_R // windows support
+struct tm *localtime_r(time_t *_clock, struct tm *_result)
+{
+  struct tm *p = localtime(_clock);
+
+  if (p)
+    *(_result) = *p;
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/GameObject/GameObject.cpp]: brackets please

**Comment by Kitzunu** [comment]
> @Kitzunu
> I ported the TC code to AC, but I am not sure why TC still send a uint32() to the client instead of a Int32().
> Check CalendarHandler.cpp, look at `holiday->CalendarFilterType`.
> I didn't touch it because probably the client does not support a negative value for this.

It should not matter

**Comment by Kitzunu** [comment]
After latest commit we got this
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK2005	"struct tm * __cdecl localtime_r(__int64 const *,struct tm *)" (?localtime_r@@YAPEAUtm@@PEB_JPEAU1@@Z) already defined in game.lib(GameEventMgr.obj)	worldserver	E:\build-azeroth\src\server\worldserver\common.lib(Util.obj)	1	
Error	LNK1169	one or more multiply defined symbols found	worldserver	E:\build-azeroth\bin\RelWithDebInfo\worldserver.exe	1	
```
@Helias

**Comment by Helias** [comment]
@Kitzunu I am still doing some tests, I don't really understand how to set localtime_r for windows only without any conflicts/error.
In TC it is added in Util.cpp, nothing else, but actually it does not work

**Comment by Kitzunu** [comment]
@Helias Latest commit fixes it :)

**Comment by Kitzunu** [comment]
========== Build: 16 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========

**Comment by felwyn** [comment]
Locked and Loaded! @Kitzunu had the fix and it worked properly! Thanks 👍

**Review by Kitzunu — APPROVED** [review]

File: src/server/game/Events/GameEventMgr.cpp [hunk 3]
```diff
@@ -23,6 +23,7 @@
 #include "LuaEngine.h"
 #endif
 #include <time.h>
+#include <Util.h>
```
Comment (Winfidonarleyan) [inline src/server/game/Events/GameEventMgr.cpp:26]:
```cpp
#include "Util.h"
```

Comment (Kitzunu) [reply src/server/game/Events/GameEventMgr.cpp:26]: need to be <util.h>

**Comment by Kitzunu** [comment]
https://github.com/azerothcore/azerothcore-wotlk/commit/e15a4939275e14b4d8ee2df94049edb34d57954e gave me a headache for this xD

**Comment by Helias** [comment]
Really thank you @Kitzunu  and @Winfidonarleyan

**Comment by FrancescoBorzi** [comment]
@Helias conflicts

**Pull request merged**